File: models/Gpt-3.5-turbo/py/code/additional_problems/Coding_Company.py

MOD = int(1e9) + 7

def solve(n, x, skills):
    dp = [[0] * (x + 1) for _ in range(n + 1)]
    dp[0][0] = 1

    for i in range(1, n + 1):
        min_skill = max_skill = skills[i - 1]
        for j in range(i, 0, -1):
            min_skill = min(min_skill, skills[j - 1])
            max_skill = max(max_skill, skills[j - 1])
            for k in range(max_skill - min_skill, x + 1):
                dp[i][k] = (dp[i][k] + dp[j - 1][k - (max_skill - min_skill)]) % MOD

        for j in range(x + 1):
            dp[i][j] = (dp[i][j] + dp[i - 1][j]) % MOD

    return sum(dp[n]) % MOD

# Read input
n, x = map(int, input().split())
skills = list(map(int, input().split()))

# Solve the problem
result = solve(n, x, skills)

# Print the result
print(result)

File: models/Gpt-3.5-turbo/cpp/code/problems/additional_problems/Coding_Company.cpp

#include <iostream>
#include <vector>
using namespace std;

const int MOD = 1000000007;

int main() {
    int n, x;
    cin >> n >> x;

    vector<int> t(n);
    for (int i = 0; i < n; i++) {
        cin >> t[i];
    }

    vector<vector<int>> dp(n + 1, vector<int>(n + 1));
    dp[0][0] = 1;

    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= i; j++) {
            int max_skill = t[i - 1], min_skill = t[i - 1];
            for (int k = i - 1; k >= 0; k--) {
                max_skill = max(max_skill, t[k]);
                min_skill = min(min_skill, t[k]);

                if (x - (max_skill - min_skill) < 0) {
                    break;
                }

                dp[i][j] = (dp[i][j] + dp[k][j - 1]) % MOD;
            }
        }
    }

    int answer = 0;
    for (int j = 1; j <= n; j++) {
        answer = (answer + dp[n][j]) % MOD;
    }

    cout << answer << endl;

    return 0;
}
